### PR TITLE
[hdkey] Consolidate and add testing for seed phrase derivation

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Cleans up and adds proper testing for seed phrase derivation.  This also opens up the path for
derivation of other keys.

Additionally, this removes `isValidPath` from the Account class, as it doesn't really belong there, but instead it is separate on it's own.

### Test Plan
Added testing for valid and invalid seed phrases, ensured that derivation changes based on the path, and that the key derivation is correct.

### CODE REVIEW

File: src/utils/hdKey.ts [hunk 1]
```diff
@@ -1,79 +1,104 @@
...
+ * Aptos derive path is 637
+ *
+ * See https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+ */
+export const APTOS_PATH_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+'\/[0-9]+'?$/;
 
-const replaceDerive = (val: string): string => val.replace("'", "");
-
-const HMAC_KEY = "ed25519 seed";
+export const ED25519_KEY = "ed25519 seed";
```
Comment (0xmaayan) [inline src/utils/hdKey.ts:20]: do we also need support for Secp256k1

**Review by 0xmaayan — APPROVED** [review]
thoughts on renaming hdKey file? have no suggestion but hdKey doesnt really mean anything to me lol

**Comment by gregnazario** [comment]
> thoughts on renaming hdKey file? have no suggestion but hdKey doesnt really mean anything to me lol

Yeah, I also was very confused until I looked it up, it's hierarchical derivation keys.  Though, its painfully long of a name, so I think we'll keep the file name to hd.

File: src/utils/hdKey.ts (hunk 1 above)
Comment (gregnazario) [reply src/utils/hdKey.ts:20]: We may but, I'll have to do more research into what the output value needs to be from it.

**Pull request merged**